src/cli/bin.ts

```diff
@@ -55,36 +108,54 @@ async function main() {
     throw new Error(`File doesn't contain .endpoints.tinyspec or .models.tinyspec extension`);
   });
 
-  const resultDefinitions: TypeDefinitionNode[] = [];
-
-  const parsingError: AnySpecError[] = [];
-  for (const source of sources) {
-    try {
-      const doc = parse(source);
-      resultDefinitions.push(...doc.definitions);
-    } catch (error) {
-      if (error instanceof AnySpecError) {
-        parsingError.push(error);
-      } else {
-        console.error('Unknown error during parsing', error);
-        processingSpinner.fail();
-        process.exit(1);
-      }
-    }
-  }
+  return sources;
+}
 
-  if (parsingError.length > 0) {
-    for (const e of parsingError) {
-      console.error(printCliError(printError(e)));
-    }
-  } else {
-    const doc = { kind: ASTNodeKind.DOCUMENT, definitions: resultDefinitions };
-    const schema = new AnySpecSchema({ ast: doc });
+function groupSourcesByNamespaces({
+  sources,
+  namespaces,
+  commonNamespace,
+}: {
+  sources: Source[];
+  namespaces: string[];
+  commonNamespace: string;
+}): Source[][] {
+  const commonRegexp = new RegExp(`\.${commonNamespace}\.(models|endpoints)\.tinyspec$`);
+
+  const namespacesRegexps = namespaces.map(
+    (n) => new RegExp(`\.${n}\.(models|endpoints)\.tinyspec$`),
+  );
+  const commonSources = sources.filter((s) => commonRegexp.test(s.name));
 
-    const errors = validate(schema, doc, baseRules);
-    errors.forEach((e) => console.error(printCliError(printError(e))));
-  }
-  processingSpinner.succeed();
+  const namespaceSources = namespacesRegexps.map((regexp) =>
+    sources.filter((s) => regexp.test(s.name)),
+  );
+
+  return namespaceSources.map((sourceArray) => sourceArray.concat(commonSources));
 }
 
-main();
+function getGroupedDocuments(groupedSources: Source[][], errCb: (err: Error) => void) {
```

Comment:
@aimmlegate  please review this code carefully and watch original source codes. IDK maybe I do some typos in mapping of 2 dimensional arrays